rest-express/src/utils/middlewares.ts

```diff
@@ -0,0 +1,41 @@
+import { BaseError } from "./errors";
+import { NextFunction, Request, Response, ErrorRequestHandler } from "express";
+import { AnyZodObject } from "zod";
+
+const middlewares: ErrorRequestHandler = (
+  err: any,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!(err instanceof BaseError)) {
+    next(err);
+  }
+
+  return res.status(err.httpCode).json({
+    error: err.description,
+    isOperational: err.isOperational,
+    additional: err.additional,
+  });
+};
+
+const validateInput =
+  (schema: AnyZodObject) =>
```

Comment:
Zod on critical path can have some performance impact. I am not sure if we should have this as a middleware in an official example or just demonstrate this in a separate endpoint.

Reply:
zod is the most widely used request validation library and it seems to be a common practice to use it. So I am assuming in any production setting this will get used anyway. We can revisit if there is a complaint about it.